### PR TITLE
chore: 更新Picker组件TypeScript声明

### PR DESCRIPTION
**Pull request by chj-damon**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
the type of `value` prop in `DatePicker` is `moment.Moment | undefined`，but the `date` param in `onChange` event is `moment.Moment | null`，the type is incompatiable.
And I find that it's not only the `DatePicker` component, but also affects `RangePicker`. 

the type definition looks like below:
```typescript
export interface SinglePickerProps {
    value?: moment.Moment;
    defaultValue?: moment.Moment;
    defaultPickerValue?: moment.Moment;
    placeholder?: string;
    renderExtraFooter?: (mode: DatePickerMode) => React.ReactNode;
    onChange?: (date: moment.Moment | null, dateString: string) => void;
}

export type RangePickerValue =
  | undefined[]
  | [moment.Moment]
  | [undefined, moment.Moment]
  | [moment.Moment, undefined]
  | [moment.Moment, moment.Moment];
```

`RangePickerValue` should support `null` also.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Picker typescript definition update         |
| 🇨🇳 Chinese | 更新Picker组件TypeScript声明          |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
请详细填一下模板。

**Comment by afc163** [comment]
这个问题应该会在 https://github.com/ant-design/ant-design/pull/20023 里彻底解决。

**Comment by chj-damon** [comment]
> 这个问题应该会在 #20023 里彻底解决。

#20023好像是给 v4 做的，但是 v3 始终还是要继续维护的嘛

**Comment by afc163** [comment]
最好写一下解决什么问题，以及提供复现的 codesandbox，修复一下 ci。

**Comment by chj-damon** [comment]
> 最好写一下解决什么问题，以及提供复现的 codesandbox，修复一下 ci。

updated.

File: components/date-picker/interface.tsx [hunk 1]
```diff
@@ -90,7 +93,7 @@ export interface RangePickerProps extends PickerProps {
   mode?: string | string[];
   separator?: React.ReactNode;
   disabledTime?: (
-    current: moment.Moment | undefined,
+    current: moment.Moment | null,
```
Comment (afc163) [inline components/date-picker/interface.tsx]: undefined 需不需要保持兼容？

Comment (chj-damon) [reply components/date-picker/interface.tsx]: 我看了一下`rc-calendar`的代码，这里的`current`的类型应该是`RangePickerValue`

**Pull request merged**